include/kwk/context/sycl/context.hpp

```diff
@@ -0,0 +1,441 @@
+//======================================================================================================================
+/*
+  KIWAKU - Containers Well Made
+  Copyright : KIWAKU Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#pragma once
+
+#include <kwk/concepts/container.hpp>
+#include <kwk/concepts/sycl_proxy.hpp>
+#include <kwk/context/sycl/proxy.hpp>
+#include <kwk/context/base.hpp>
+#include <kwk/context/cpu/context.hpp>
+#include <kwk/utility/position.hpp>
+#include <kwk/context/sycl/internal/sycl_tools.hpp>
+#include <cmath>
+#include <functional>
+
+
+
+namespace kwk::sycl
+{
+  struct context : public ::sycl::queue, public kwk::base_context<context>
+  {
+  public:
+    using parent = ::sycl::queue;
+    using parent::parent; // forwards every constructor of queue to context.
+
+    static auto in   (kwk::concepts::container auto const& c) { return kwk::sycl::in{c};    }
+    static auto out  (kwk::concepts::container auto& c)       { return kwk::sycl::out{c};   }
+    static auto inout(kwk::concepts::container auto& c)       { return kwk::sycl::inout{c}; }
+
+
+    void print_sycl_header()
```

Comment:
would be slightly better if it was

`std::ostream& print_sycl_header(std::ostrream& os = std::cout)

and we're using os inside instead of depending on cout directly